python/lsst/ts/externalscripts/maintel/tma/random_walk_and_take_image_gencam.py

```diff
@@ -0,0 +1,520 @@
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              program:
+                description: Optional name of the program this data belongs to, e.g. WFD, DD, etc.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              camera_playlist:
+                description: >-
+                  Optional name a camera playlist to load before running the script.
+                  This parameter is mostly designed to use for integration tests and is
+                  switched off by default (e.g. null).
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              max_dome_mount_az:
+                description: >-
+                  Maximum difference between dome position and target mount azimuth in degrees.
+                  If the actuall difference is larger than this parameter, the dome moves.
+                type: number
+                default: 10.
+              dome_offset:
+                description: Artificial offset for when the dome loses reference.
+                type: number
+                default: 0
+            required:
+            - total_time
+            - radius
+            - min_az
+            - max_az
+            - min_el
+            - max_el
+            - big_offset_prob
+            - big_offset_radius
+            - track_for
+            - stop_when_done
+            - ignore
+            - rot_value
+            - rot_type
+            - az_wrap_strategy
+            - camera_sal_indexes
+            - exp_times
+            - num_exp
+            - sleep_time
+            - reason
+            - program
+            - camera_playlist
+            - max_dome_mount_az
+            - dome_offset
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def load_playlist(self):
+        """Load playlist."""
+        raise NotImplementedError()
+
+    async def move_dome(self, az: float):
+        """Move the dome to a give position
+
+        Parameters
+        ----------
+        az : float
+            New dome Azimuth position in degrees.
+        """
+        # Return right away if ignoring the dome
+        if "mtdome" in self.config.ignore:
+            return
+
+        # The dome usually goes to a fault when the brakes engage.
+        # This usually happens because they are engaged when the dome
+        # is slighly out of its target position. For this script,
+        # this difference is negligible.
+        await self.tcs.rem.mtdome.cmd_exitFault.set_start()
+
+        # This time is required to have the Dome getting out its Fault
+        # state before moving
+        await asyncio.sleep(5)
+
+        # Determine from the current position if we want to move or not
+        dome_az_encoder = await self.tcs.rem.mtdome.tel_azimuth.next(flush=True)
```

Comment:
I notice here and in several other calls to `next` and `aget` that you are not specifying a timeout. This could potentially cause a deadlock if the telemetry is not being published. Please, make sure you do a global search to all these occurrences and add a timeout.

Reply:
I **think** I addressed all of them.